sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -51,6 +51,31 @@ class RegexParser(pattern: String) {
     ast
   }
 
+  def parseReplacement(numCaptureGroups: Int): RegexReplacement = {
+    val sequence = RegexReplacement(new ListBuffer(), numCaptureGroups)
+    while (!eof()) {
+      parseReplacementBase(() => eof()) match {
```

Comment:
remove unused param:

```suggestion
      parseReplacementBase() match {
```